File: objective-c/iOS13Display/iOS13Display/Roots/RootUIView.m
//
//  RootUIView.m
//  iOS13Display
//
//  Created by Alexandr Kirilov on 08/12/2019.
//  Copyright © 2019 Alexandr Kirilov. All rights reserved.
//

#import "RootUIView.h"

@interface RootUIView()

-(void) mSetupView;

@end

@implementation RootUIView

-(instancetype) initWithFrame:(CGRect)frame {
	
	self = [super initWithFrame:frame];
	if (self) {
		[self mSetupView];
	}
	return self;
}

-(id) initWithCoder:(NSCoder *)coder {
	
	self = [super initWithCoder:coder];
	if (self) {
		[self mSetupView];
	}
	return self;
}

-(void) mSetupView {
	
	if (@available(iOS 11,*)) {
		self.accessibilityIgnoresInvertColors = YES;
	}
	if (@available(iOS 13,*)) {
		self.overrideUserInterfaceStyle = UIUserInterfaceStyleLight;
	}
}

@end

File: swift/iOS13Display/iOS13Display/Roots/RootUIView.swift
//
//  RootUIView.swift
//  iOS13Display
//
//  Created by Alexandr Kirilov on 08/12/2019.
//  Copyright © 2019 Alexandr Kirilov. All rights reserved.
//

import UIKit

class RootUIView: UIView {
	
	convenience init() {
		
		self.init(frame: CGRect.zero);
	}
	
	override init(frame: CGRect) {
		
		super.init(frame: frame);
		if #available(iOS 13.0, *) {
			self.overrideUserInterfaceStyle = .light;
		}
		if #available(iOS 11.0, *) {
			self.accessibilityIgnoresInvertColors = true;
		}
	}
	
	required init(coder: NSCoder) {
		
		fatalError("init(coder:) has not been implemented");
	}
}
